document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java

```diff
@@ -40,69 +36,42 @@
 
     private static final String SMALL_FULL = "smallfull";
 
+    private static final String ABRIDGED_REGEX = "/transactions\\/[0-9-]+/accounts\\/.*";
+
+    private static final String SMALL_FULL_REGEX = "/transactions\\/[0-9-]+/company-accounts\\/.*";
+
     private static final String ERROR_CALLING_ACCOUNTS = "An error occurred when calling: %s handler to obtain: %s" +
-             " accounts data for transaction: %s and resource link: %s";
+             " accounts data for resourceUri: %s";
 
     @Override
     public DocumentInfoResponse getDocumentInfo(DocumentInfoRequest documentInfoRequest) throws DocumentInfoException {
 
-        String resourceId = documentInfoRequest.getResourceId();
         String resourceUri = documentInfoRequest.getResourceUri();
         String requestId = documentInfoRequest.getRequestId();
 
         final Map< String, Object > debugMap = new HashMap< >();
         debugMap.put("resource_uri", resourceUri);
-        debugMap.put("resource_id", resourceId);
 
         LOG.infoContext(requestId,"Started getting document info", debugMap);
 
-        Transaction transaction;
-        try {
-            transaction = transactionService.getTransaction(resourceId, requestId);
-        } catch (ServiceException e) {
-            LOG.errorContext(requestId,"An error occurred when calling the transaction service with resource id: "
-                    + resourceId, e, debugMap);
-            throw new DocumentInfoException("Failed to get transaction with resourceId: " + resourceId, e);
-        }
-
-        String resourceLink =  Optional.of(transaction)
-                .map(Transaction::getResources)
-                .map(resources -> resources.get(resourceUri))
-                .map(Resources::getLinks)
-                .map(links -> links.get(LinkType.RESOURCE.getLink()))
-                .orElseGet(() -> {
-                    LOG.infoContext(requestId,"Unable to find resource: " + resourceId
-                            + " in transaction: " + resourceUri, debugMap);
-                    return "";
-                });
-
-
-        // Will need to be refactored when additional accounts are added.
         String accountType = "";
         try {
-            if (isAbridgedAccounts(resourceLink)) {
+            if (resourceUri.matches(ABRIDGED_REGEX)) {
                 accountType = ABRIDGED;
-                return abridgedAccountsDataHandler.getAbridgedAccountsData(transaction, resourceLink, requestId);
-            } else {
+                return abridgedAccountsDataHandler.getAbridgedAccountsData(resourceUri, requestId);
+
+            } else if (resourceUri.matches(SMALL_FULL_REGEX)) {
                 accountType = SMALL_FULL;
-                return smallFullAccountsDataHandler.getSmallFullAccountsData(transaction, resourceLink, requestId);
+                return smallFullAccountsDataHandler.getSmallFullAccountsData(resourceUri, requestId);
```

Comment:
How is this scalable to other account types? The small-full handler is being used when a company accounts URL is matched. New account types will also be within a company accounts?

Reply:
I'll rename the handler to companyAccounts and add a new implementation to deal with the retrieval of the small full data

Reply:
completed